### PR TITLE
Fix assembly version attributes for C#

### LINKED ISSUE #6201: Apply AssemblyFileVersion/AssemblyInformationalVersion attributes on assembly
(For .NET.)

Our AssemblyVersion attribute should *either* be just the major version number (as per https://codingforsmarties.wordpress.com/2016/01/21/how-to-version-assemblies-destined-for-nuget/) or it should be major.minor (but not patch), to avoid requiring binding redirects.

More information should be applied in the AssemblyFileVersion and AssemblyInformationalVersion attributes. For example, closer to release we might have `AssemblyFileVersion("1.0.0.0")` and `AssemblyInformationalVersion("1.0.0-beta1")`.

### PR DESCRIPTION
**Pull request by jtattermusch**
Fixes #6201 (thanks for reporting it!).

### CODE REVIEW

File: src/csharp/Grpc.Core/VersionInfo.cs [hunk 1]
```diff
@@ -35,13 +35,20 @@ namespace Grpc.Core
...
+    /// for rationale about assembly versioning.
     /// </summary>
     public static class VersionInfo
     {
         /// <summary>
-        /// Current version of gRPC C# assemblies
+        /// Current <c>AssemblyVersion</c> attribute of gRPC C# assemblies
         /// </summary>
-        public const string CurrentAssemblyVersion = "0.14.0.0";
+        public const string CurrentAssemblyVersion = "1.0.0.0";
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core/VersionInfo.cs]: The major version now is technically "0", but does it make sense to make the assembly version "1.0.0.0", as that would avoid the need for redirect once GA is released?

**Comment by jtattermusch** [comment]
@jskeet, can you please review this small change?

**Comment by jskeet** [comment]
LGTM. Will there be other PRs for other assemblies, or do they determine their versions from Grpc.Core?

**Comment by jtattermusch** [comment]
Thanks for reviewing. 
The Version.cs file is link-included by all other projects and grabs the data from VersionInfo.cs file, so no other changes are needed.

**Comment by jtattermusch** [comment]
grpc-mac-node3 is failure is not related.

**Pull request merged**